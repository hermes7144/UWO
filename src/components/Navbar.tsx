import { Link } from 'react-router-dom';
import { BsFillPencilFill } from 'react-icons/bs';
import User from './User';
import Button from './ui/Button';
import { useAuthContext } from './context/AuthContext';
import { login, logout } from '../api/firebase';

export default function Navbar() {
  const { user } = useAuthContext();

  return (
    <header className='flex justify-between border-b border-gray-300 p-2'>
      <Link to='/' className='flex items-center text-4xl text-brand'>
        <h1>UWOInsight</h1>
      </Link>
      <nav className='flex items-center gap-4 font-semibold'>
        {user?.isAdmin && (
          <Link to='/items/new'>
            <BsFillPencilFill />
          </Link>
        )}

        {user && <User user={user} />}
        {!user && <Button text={'Login'} onClick={login}></Button>}
        {user && <Button text={'Logout'} onClick={logout}></Button>}
      </nav>
    </header>
  );
}
