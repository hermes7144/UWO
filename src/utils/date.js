// import { format, register } from 'timeago.js';
// import koLocale from 'timeago.js/lib/lang/ko';

// register('ko', koLocale);

// export function formatAgo(date, lang = 'en_US') {
//   return format(date, lang);
// }

export function formatDate(rawDate) {
  const date = new Date(rawDate);
  const year = date.getFullYear().toString().slice(-2); // 연도에서 뒤의 두 자리만 추출
  const month = ('0' + (date.getMonth() + 1)).slice(-2); // 월을 두 자리 숫자로 변환
  const day = ('0' + date.getDate()).slice(-2); // 일을 두 자리 숫자로 변환
  return `${year}/${month}/${day}`;
}