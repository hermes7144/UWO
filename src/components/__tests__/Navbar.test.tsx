import { render, screen } from '@testing-library/react';
import Navbar from '../Navbar';
import { withAuthContexts, withRouter } from '../../tests/utils';
import { Route } from 'react-router-dom';

describe('Navbar', () => {
  let user = null;
  afterEach(() => {
    user = null;
  });

  test('renders user is empty', () => {
    user = { displayName: 'test' };
    const { asFragment } = renderNavbar();

    expect(asFragment()).toMatchSnapshot();
  });

  test('renders user is not login', () => {
    renderNavbar();

    const navbarElement = screen.getByText('Login');
    expect(navbarElement).toBeInTheDocument();
  });

  function renderNavbar() {
    return render(withAuthContexts(withRouter(<Route path='/' element={<Navbar />} />), user));
  }
});
