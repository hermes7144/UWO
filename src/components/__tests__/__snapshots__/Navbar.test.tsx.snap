// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders user is empty 1`] = `
<DocumentFragment>
  <header
    class="flex justify-between border-b border-gray-300 p-2"
  >
    <a
      class="flex items-center text-xl font-bold text-brand"
      href="/"
    >
      <h1>
        UWO TOOL
      </h1>
    </a>
    <nav
      class="flex items-center gap-4 font-semibold"
    >
      <a
        href="/routes"
      >
        <h2>
          무역 루트
        </h2>
      </a>
      <div
        class="flex items-center shrink-0"
      >
        <img
          alt="test"
          class="w-10 h-10 rounded-full mr-2"
        />
        <span
          class="hidden md:block"
        >
          test
        </span>
      </div>
      <button
        class="bg-brand text-white
 py-2 px-4 rounded-sm hover:brightness-110"
      >
        Logout
      </button>
    </nav>
  </header>
</DocumentFragment>
`;
